tests/foreman/ui/test_reporttemplates.py

```diff
@@ -345,3 +403,60 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
     :expectedresults: Report can't be downloaded. Error.
     :CaseImportance: High
     """
+
+
+@tier3
+def test_positive_generate_entitlements_reports_multiple_formats(session, content_setup):
+    """Generate reports using the Entitlements template in html, yaml, json, and csv format.
+
+        :id: b268663d-c213-4e59-8f81-61bec0838b1e
+
+
+        :setup: Installed Satellite with Organization, Activation key,
+                Content View, Content Host, and Subscriptions.
+
+        :steps:
+            1. Monitor -> Report Templates
+            2. Entitlements -> Generate
+            3. Click the dropdown to select output format
+            4. Submit
+
+        :expectedresults: reports are generated containing all the expected information
+                          regarding Entitlements for each output format.
+
+        :CaseImportance: High
+    """
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+        vm.install_katello_ca()
+        vm.register_contenthost(content_setup[0].label, content_setup[1].name)
+        assert vm.subscribed
+        with session:
+            session.organization.select(content_setup[0].name)
+            session.location.select('Default Location')
+            result_json = session.reporttemplate.generate("Entitlements",
+                                                          values={'output_format': 'JSON'})
+            with open(result_json) as json_file:
+                data_json = json.load(json_file)
+                assert data_json[0]['Name'] == vm.hostname
+                assert data_json[0]['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+            result_yaml = session.reporttemplate.generate("Entitlements",
+                                                          values={'output_format': 'YAML'})
+            with open(result_yaml) as yaml_file:
+                data_yaml = yaml.load(yaml_file, Loader=yaml.FullLoader)
+                assert data_yaml[0]['Name'] == vm.hostname
```

Comment:
Are you 100% certain our hostname will be in very first line? Another test that created and registered host might be running in parallel.

Perhaps something like that would be more robust:
```python
assert any(entitlement['Name'] == vm.hostname for entitlement in data_yaml)
```

Same applies to all other formats.

Reply:
Please use `any` also for second assertion:
```python
            assert any(entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME for entitlement in data_json)
```

That applies to all formats

Reply:
this assertion is too long when flake8 gets run. Should I rewrite it as a for loop instead of using list comprehension?

Reply:
You can break comprehension into multiple lines:
```python
            assert any(entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
                       for entitlement in data_json)
```
```python
            assert any(
                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
                for entitlement in data_json
            )
```
If flake is still unhappy, perhaps short variable name than `entitlement` would be in place.

You can also assign comprehension to new variable and assert on that (maybe with better variable name than `result`), but it won't make much difference:
```python
            result = [
                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
                for entitlement in data_json)
            ]
            assert any(result)
```

Personally I think tools are here to support us and something is wrong when we spent resources on making tool happy, so do anything that makes flake stop complaining and let's move to more important things.